### PR TITLE
Add an initial Go SDK Roadmap

### PR DESCRIPTION
**Pull request by lostluck**
Added my current concerns and plans for working on the SDK. 

R: @aaltay @herohde 

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by lostluck** [comment]
R: @herohde @aaltay Ping!

File: website/src/roadmap/go-sdk.md [hunk 1]
```diff
@@ -20,7 +20,53 @@ limitations under the License.
...
+## Usability
+
+The Go SDK should not block users from writing the pipelines they need to run.
+
+* Support large elements [BEAM-4124](https://issues.apache.org/jira/browse/BEAM-4124)
+* Coder Registry to support complex user elements [BEAM-3306](https://issues.apache.org/jira/browse/BEAM-3306)
+
+## Performance
+
+Since the Go SDK has no official runner support at this time, this gives us the chance to work the performance
```
Comment (aaltay) [inline website/src/roadmap/go-sdk.md]: `work the` -> `work on the`

File: website/src/roadmap/go-sdk.md [hunk 2]
```diff
@@ -20,7 +20,53 @@ limitations under the License.
...
+
+Without SDFs, IOs are always constrained to work on a single machine prior to a sharding operation (like CoGroupByKey),
+which makes scalable IOs difficult to impossible to write.
+
+Otherwise, improving examples and documentation for devs and users alike is ongoing. 
+Contributions are welcome. Please contact the [dev list](mailto:dev@beam.apache.org?subject=%5BGo%20SDK%5D%20How%20can%20I%20help%3F)
+for assistance in finding a place to help out.
+
+ - JIRA: [sdk-go](https://issues.apache.org/jira/issues/?jql=project%20%3D%20BEAM%20AND%20component%20%3D%20sdk-go) / [BEAM-2083](https://issues.apache.org/jira/browse/BEAM-2083)
+ - Contact: Robert Burke (@lostluck) [Email](mailto:rebo@google.com,dev@beam.apache.org?subject=%5BGo%20SDK%5D)
```
Comment (aaltay) [inline website/src/roadmap/go-sdk.md]: Do you want to put your email on the website?

Consider just limiting the dev@ link with the subject line. (it is your call)

Comment (lostluck) [reply website/src/roadmap/go-sdk.md]: That's intentional. I don't look at the dev list every day, but I'm certainly looking at my work email during business hours, and I'm interested in a ping for Go SDK inquiries.

For everything else, there's spam filters.

**Comment by lostluck** [comment]
@aaltay  Thanks for the review! Please merge.

**Pull request merged**